### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by gedewon**
# PR checklist : 
- [x] Use preprocessed HTML file with embedded Ruby code.
- [x] Use layouts and templates for shared content.

### CODE REVIEW

**Review by descholar-ceo — APPROVED** [review]
Hi @Gedewon 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all requirements ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by peterrobert — COMMENTED** [review]
### INVALID

- [x] It seems you have submitted a pull request that has already been approved. Because of this, I can not continue to review your project.

**Comment by gedewon** [comment]
@peterrobert  yes I am repeating the module, that's why it's already approved and merged.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::Base
+  def current_user
+    @first_user = User.first
+  end
```
Comment (mateo951) [inline app/controllers/application_controller.rb:4]: ApplicationController contains #current_user method ✔

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -6,5 +6,30 @@ def index
...
+    post.save
+    # post.update_posts_counter
+    if post.save
+      # puts "Saved successfully"
+      redirect_to user_path(current_user)
+    else
+      # puts "Failed to save"
+      redirect_to new_user_post_path
+    end
+  end
```
Comment (mateo951) [inline app/controllers/posts_controller.rb:30]: A post is created on behalf of the currently logged-in user (current_user) of the app 🔥 Fantastic job!

File: config/routes.rb [hunk 3]
```diff
@@ -4,6 +4,9 @@
   # Defines the root path route ("/")
   root "users#index"
   resources :users, only: [:index, :show] do
-    resources :posts ,only: [:index,:show]
+    resources :posts ,only: [:index,:show,:new, :create] do
```
Comment (mateo951) [inline config/routes.rb:7]: Routes in the config/routes.rb file for creating posts ✔

File: app/views/posts/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      </div>
+ </div>
+ <%= form_with model: @post, url: user_posts_path do |f| %>
+      <%= f.label :title %><br>
+      <%= f.text_field :title %><br>
+      <%= f.label :text %><br>
+      <%= f.text_area :text %><br>
+      <%= f.submit "Create Post" %>
+    <% end %>
+<div>
```
Comment (mateo951) [inline app/views/posts/new.html.erb:17]: There is a form in the views/posts/new.html.erb file which accepts input for creating posts 🔥

File: app/controllers/comments_controller.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    puts parameters
+    comment = Comment.new(post_id: params[:post_id], user_id: current_user.id, text: parameters[:text])
+    comment.save
+
+    if comment.save
+      redirect_to user_post_path(id: params[:post_id])
+    else
+      redirect_to new_user_post_comment
+    end
+  end
```
Comment (mateo951) [inline app/controllers/comments_controller.rb:19]: A comment is created on behalf of the currently logged-in user (current_user) of the app 👌 Well done!

**Review by mateo951 — COMMENTED** [review]
### Hi @Gedewon, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 

### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**